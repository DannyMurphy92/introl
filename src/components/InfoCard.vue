<template>
  <div class="info-card">
    <div class="info-card__header">
      <span>{{ header }}</span>
    </div>
    <div class="info-card__body">
      <span>{{ body }}</span>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'InfoCard',

  props: {
    header: {
      type: String,
      required: true,
    },
    body: {
      type: String,
      required: true,
    },
  },
});
</script>

<style lang="scss">
.info-card {
  width: 300px;

  &__header {
    font-size: 24px;
    font-weight: $font-weight-bold;
    background: $color-grey;
    border-radius: 3px;
    padding: 10px 0;
    margin-bottom: 15px;
  }
}
</style>
