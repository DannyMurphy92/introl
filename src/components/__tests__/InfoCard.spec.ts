import InfoCard from '../InfoCard.vue';
import { shallowMount, Wrapper } from '@vue/test-utils';

describe('InfoCard', () => {
  let wrapper: Wrapper<any>;

  const propsData = {
    header: 'Test header',
    body: 'Test body',
  };

  const mountComponent = () => {
    wrapper = shallowMount(InfoCard, { propsData });
  };
  describe('GIVEN initial state', () => {
    beforeEach(() => {
      mountComponent();
    });

    it('THEN renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
