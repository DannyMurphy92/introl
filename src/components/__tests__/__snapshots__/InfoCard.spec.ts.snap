// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard GIVEN initial state THEN renders correctly 1`] = `
<div class="info-card">
  <div class="info-card__header"><span>Test header</span></div>
  <div class="info-card__body"><span>Test body</span></div>
</div>
`;
